crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -1027,8 +1036,29 @@ where
                         self.try_make_sync_target_canonical();
                     }
                     BlockStatus::Disconnected { missing_parent } => {
-                        // continue downloading the missing parent
-                        self.sync.download_full_block(missing_parent.hash);
+                        // get the latest valid finalized block or the tip (if we don't have the
+                        // finalized block set)
+                        let fork_ancestor_num = self.forkchoice_state_tracker.last_valid_finalized().and_then(|finalized| {
+                            // now we need to actually fetch these
+                            self.blockchain.block_number(finalized).unwrap_or_else(|err| {
+                                // TODO: store full headers (or BlockNumHash) in forkchoice
+                                // state tracker if valid, so we don't need to fetch at all
+                                error!(target: "consensus::engine", ?finalized, ?err, "Failed to get block number from validated finalized block hash");
+                               None
+                            })
+                        }).unwrap_or_else(|| self.blockchain.canonical_tip().number);
+
```

Comment:
It's not immediately clear to me what we're checking here.
what exactly are we checking here?

this looks way too complex

this should be much easier imo

Reply:
I thought the latest valid finalized block would be give a better heuristic for how long the new fork could be, but yeah it's too complex and using the head is much simpler, so switched to doing that